nemo/collections/tts/g2p/models/ctc.py

```diff
@@ -125,10 +125,14 @@ def setup_grapheme_tokenizer(self, cfg):
                 punctuation_marks = string.punctuation.replace('"', "").replace("\\", "").replace("'", "")
                 chars += punctuation_marks
 
-            vocab_file = "/tmp/char_vocab.txt"
-            with open(vocab_file, "w") as f:
-                [f.write(f'"{ch}"\n') for ch in chars]
-                f.write('"\\""\n')  # add " to the vocab
+            if cfg.tokenizer_grapheme.vocab_file:
+                vocab_file = cfg.tokenizer_grapheme.vocab_file
+            else:
+                vocab_file = "/tmp/char_vocab.txt"
+
+                with open(vocab_file, "w") as f:
```

Comment:
we actually do need to have this for a proper model restoration [tests are failing], could you move lines 133-135 outside of `else`?

Reply:
shouldn't restoration work as the tokenizer_grapheme.vocab_file is not set ? i will look into it,
but just moving the lines out of if file would make the if do nothing.